### PR TITLE
Fix tile visual bugs: overflow and spacing

### PR DESCRIPTION
**Pull request by mayank99**
- Fixed tile name truncation using min-width on .iui-content.
- Fixed metadata not staying at the bottom using margin-top: auto.
- Fixed metadata hiding even in absence of options menu by using `last-child` selector.
- Fixed IE fallback in `iui-line-clamp` mixin. *IE does not support `@supports`.* 😄

### CODE REVIEW

**Comment by mayank99** [comment]
This is the diff from test images:
image

File: src/tile/tile.scss [hunk 1]
```diff
@@ -175,9 +175,7 @@
       }
 
       > .iui-name-label {
-        text-overflow: ellipsis;
-        overflow: hidden;
-        white-space: nowrap;
+        @include iui-line-clamp(1);
```
Comment (bentleyvk) [inline src/tile/tile.scss:178]: Can `iui-line-clamp` be used here? 
image

Comment (bentleyvk) [inline src/tile/tile.scss:178]: I think we need `@supports` with a fallback to ellipsis for IE.

File: backstop/tests/tile.html [hunk 2]
```diff
@@ -568,6 +559,21 @@ <h2>Default</h2>
               />
             </svg>
           </button>
+
+          <div class="iui-metadata">
```
Comment (bentleyvk) [inline backstop/tests/tile.html:563]: Why you need to switch places?

File: src/tile/tile.scss [hunk 3]
```diff
@@ -204,6 +202,7 @@
       width: 100%;
       display: flex;
       align-items: center;
+      margin-top: auto;
```
Comment (bentleyvk) [inline src/tile/tile.scss:205]: Is it needed?

File: backstop/tests/tile.html (hunk 2 above)
Comment (veekeys) [reply backstop/tests/tile.html:563]: Yeah, looks really weird, knowing that metadata should go before options menu...

File: src/tile/tile.scss (hunk 1 above)
Comment (mayank99) [reply src/tile/tile.scss:178]: the mixin already has that

File: backstop/tests/tile.html (hunk 2 above)
Comment (mayank99) [reply backstop/tests/tile.html:563]: it's either this or I can use `:last-child`

But right now, the metadata width decreases on hover even when there's no options menu

File: src/tile/tile.scss (hunk 3 above)
Comment (mayank99) [reply src/tile/tile.scss:205]: Yeah, read description

File: src/tile/tile.scss (hunk 1 above)
Comment (mayank99) [reply src/tile/tile.scss:178]: Done

File: backstop/tests/tile.html (hunk 2 above)
Comment (mayank99) [reply backstop/tests/tile.html]: I changed to `last-child` but it won't work when user has custom children (which get added after metadata and more options)

File: src/tile/tile.scss (hunk 1 above)
Comment (bentleyvk) [inline src/tile/tile.scss:178]: Having `min-width: 0` on `iui-content` does the same and also works on IE.

Comment (mayank99) [reply src/tile/tile.scss:178]: How does that even work?

Comment (mayank99) [reply src/tile/tile.scss:178]: I changed the line-clamp mixin with proper fallback so it should be fixed now

Comment (bentleyvk) [reply src/tile/tile.scss:178]: When using `flex` by default `min-width` is the same size as its content therefore it goes outside the bounds.
Maybe I am old fashioned but for single line ellipsis I still prefer `text-overflow: ellipsis` 😀

Comment (mayank99) [reply src/tile/tile.scss]: There, this should satisfy your old fashioned needs 😄

**Review by bentleyvk — APPROVED** [review]

**Pull request merged**